### ISSUE TITLE
Why UpSampling instead of Deconvolution 

### ISSUE THREAD

**Issue by iamharsha1999** [pvitoria/ChromaGAN#5]
First of all thanks for your paper and the official implementation in Keras. Last week, I was trying to implement the paper in PyTorch. While doing the implementation I observed that you have used Upsampling in the below part of your code. Out of curiosity I just wanted to ask you that why you went for Upsampling rather than Deconvolution. Upsampling is just an interpolation method it has no learnable parameters If you had used Deconvolution in its place wouldn't have given much better results as it would have adjusted its parameters while training. I'm a beginner and I'm just asking this out of curiosity. Pls, do correct me if I'm wrong somewhere.
        
        # Fusion + Colorization
        outputModel =  keras.layers.Conv2D(256, (1, 1), padding='same', strides=(1, 1), activation='relu')(modelFusion)
        outputModel =  keras.layers.Conv2D(128, (3, 3), padding='same', strides=(1, 1), activation='relu')(outputModel)

        outputModel =  keras.layers.UpSampling2D(size=(2,2))(outputModel)
        outputModel =  keras.layers.Conv2D(64, (3, 3), padding='same', strides=(1, 1), activation='relu')(outputModel)
        outputModel =  keras.layers.Conv2D(64, (3, 3), padding='same', strides=(1, 1), activation='relu')(outputModel)

        outputModel =  keras.layers.UpSampling2D(size=(2,2))(outputModel)
        outputModel =  keras.layers.Conv2D(32, (3, 3), padding='same', strides=(1, 1), activation='relu')(outputModel)
        outputModel =  keras.layers.Conv2D(2, (3, 3), padding='same', strides=(1, 1), activation='sigmoid')(outputModel)
        outputModel =  keras.layers.UpSampling2D(size=(2,2))(outputModel)
        final_model = Model(input=input_img, outputs = [outputModel, global_featuresClass])

**Comment by GiantKevin** [comment]
Mayve you can read this paper  —— **Deconvolution** and Checkerboard Artifacts .
